### DISCUSSION TITLE
Snap layout suggestions

### DISCUSSION THREAD

**Discussion by ghost** [rxhanson/Rectangle#985] [Ideas]
1) Consider a snap layout feature when hovering over the green stop light button, like e.g., in Windows 11 
![Windows-11-Snap-Layout-large-display-1033913632](https://user-images.githubusercontent.com/72652159/199844012-20482507-9f9f-4b98-ad9f-7e8cebedb060.jpg)
(Might be related to [https://github.com/rxhanson/Rectangle/discussions/446](url))

2) Consider displaying a snap layout suggestion when moving a window to the top of the screen. The window can be dragged directly to a selection of pre-made snap layouts (preferably customizable :). When letting the window go on a selected snap layout the window resizes to that layout. [https://youtu.be/FqHhgfuQG6w](url)

**Comment by rxhanson** [comment]
Thanks for the feedback. I don't have plans to add this into Rectangle, but see no reason to turn down a good pull request for anything you've mentioned.

My implementation of a "Snap Panel" is in [Rectangle Pro](https://rectangleapp.com/pro), in the Snap Areas tab. I did not add the hover stoplight button functionality because having to move a cursor to a precise location like that is slow compared to alternative ways to snap windows, and personally I never use the stoplight buttons anyway. The Snap Panel in pro can be configured to display on different edges/corners of the screen or next to the cursor, and it can be configured with any size/position you want. You can even build your own custom shortcuts to put in there (although at the moment they all have the same icon - it's on my todo for the next release to have letters on custom shortcut icons in the snap panel).

**Reply by ghost** [reply]
Thank you for your very quick response :) 
I will try this out in Rectangle Pro :) (might i receive a rectangle pro license at a discount?)

**Reply by rxhanson** [reply]
As per the Rectangle contribution policy, Rectangle contributors get a Pro license for free (even for localizations). Just send me an email.

**Reply by ghost** [reply]
Thank you very much! 🥳I just sent you an email.
